src/routes/guides/testing-your-app.mdx

```diff
@@ -0,0 +1,404 @@
+
+- `.toBeInTheDocument()` - checks if the element actually exists in the DOM
+- `.toBeVisible()` - checks if there is no reason the element should be hidden
+- `.toHaveTextContent(content)` - checks if the text content matches
+- `.toHaveFocus()` - checks if this is the currently focused element
+- `.toHaveAccessibleDescription(description)` - checks accessible description
+- and a [lot more](https://github.com/testing-library/jest-dom?tab=readme-ov-file#custom-matchers).
+
+
+## Directive testing
+
+[Directives](/reference/jsx-attributes/use) are reusable behavior for elements. As such, they get a reference to the element and the property value given to them as an accessor. To make testing them more concise, our `@solidjs/testing-library` has a renderDirective function:
+
+
+```ts frame="none"
+const renderResult = renderDirective(directive, {
+  initialValue, // value initially added to the argument signal
+  targetElement, // opt. node name or element used as target for the directive
+  ...renderOptions, // see render options
+})
+const {
+  arg, // getter for the directive's argument
+  setArg, // setter for the directive's argument
+  ...renderResults, // see render results
+} = renderResult
+```
+
+The `container` from the `...renderResults` will contain the targetElement (default is a `<div>`). That and the ability to manipulate the `arg` signal are helpful in testing directives.
+
+If for example there is a directive that handles the Fullscreen API, it could be tested like this:
+
+<TabsCodeBlocks>
+  <div id="fullscreen.test.ts">
+```ts frame="none"
+import { test, expect, vi } from "vitest"
+import { renderDirective } from "@solidjs/testing-library"
+import { createFullScreen } from "./fullscreen"
+
+test("toggles fullscreen", () => {
+  const targetElement = document.createElement("div")
+  const fs = vi.spyOn(targetElement, "fullscreen")
+  const [setArg, container] = renderDirective(createFullScreen, false)
+  setArg(true)
+  expect(fs).toHaveBeenCalled()
+})
+```
+  </div>
+  <div id="fullscreen.ts">
+```ts frame="none"
+import { Accessor } from "solid-js"
+
+export const fullscreen = (ref: HTMLElement, active: Accessor<boolean>) =>
+  createEffect(() => {
+    const isActive = document.fullscreenElement === ref
+    if (active() && !isActive) {
+      ref.requestFullScreen().catch(() => {})
+    } else if (!active() && isActive) {
+      document.exitFullScreen()
+    }
+  })
+```
+  </div>
+</TabsCodeBlocks>
+
+## Primitive testing
+
+If the reference to the element is not needed, parts of state and logic can be put into reusable hooks or primitives. Since those do not require any elements, using `render` to test them would require a component that otherwise has no use. To circumvent that, there is a `renderHook` utility that simulates such a component without actually rendering anything.
```

Comment:
```suggestion
When the reference to an element is not needed, parts of state and logic can be put into reusable hooks or primitives. 
Since these do not require elements, there is no need for `render` to test them since it would require a component that has no other use.
To avoid this, there is a [`renderHook` utility](https://testing-library.com/docs/solid-testing-library/api#renderhook) that simulates a component without actually rendering anything.
```